### PR TITLE
Simulation improvements; Flight stats fixes; Memory allocation fixes

### LINKED ISSUE #55: Create a Flight Statistics File
Keep most important information about the flight (duration, events & actions, max altitude, max velocity, max acceleration,...) in a flight_stats file. 

An open question is when to create this file. Also care has to be taken when accessing LFS from multiple tasks.

### LINKED ISSUE #70: Add Calibration To Flight Stats file.

### PR DESCRIPTION
**Pull request by stojadin2701**
Closes #70, closes #55

### CODE REVIEW

**Review by stojadin2701 — COMMENTED** [review]
In task_recorder.c, before           

(osMessageQueueGet(rec_queue, &curr_log_elem, NULL, osWaitForever)

add:

    if (global_recorder_status < REC_FILL_QUEUE) {
              break;
    }

and don't wait forever

File: boards/cats_rev1Pro/src/flash/recorder.c [hunk 1]
```diff
@@ -142,7 +142,9 @@ void record(timestamp_t ts, rec_entry_type_e rec_type_with_id, const void *rec_v
       case FLIGHT_INFO:
         e.u.flight_info = *((flight_info_t *)rec_value);
         /* Record the flight info stats before deciding whether to record this entry or not. */
-        collect_flight_info_stats(ts, e.u.flight_info);
+        if (global_recorder_status >= REC_FILL_QUEUE) {
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/flash/recorder.c:145]: check not needed anymore

File: boards/cats_rev1Pro/src/flash/recorder.h [hunk 2]
```diff
@@ -141,11 +147,17 @@ extern flight_stats_t global_flight_stats;
...
-  memset(&global_flight_stats, 0, sizeof(flight_stats_t));
+inline void init_global_flight_stats() {
+  /* Save current flight config */
+  memcpy(&global_flight_stats.config, &global_cats_config, sizeof(global_cats_config));
+  //memset(&global_flight_stats, 0, sizeof(flight_stats_t));
   global_flight_stats.max_height.val = -INFINITY;
   global_flight_stats.max_velocity.val = -INFINITY;
   global_flight_stats.max_acceleration.val = -INFINITY;
+
+//  global_flight_stats.calibration_data.gyro_calib = (vf32_t){.0f, .0f, .0f};
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/flash/recorder.h:158]: delete those lines

File: boards/cats_rev1Pro/src/util/actions.c [hunk 3]
```diff
@@ -238,12 +239,13 @@ bool set_recorder_state(int16_t state) {
       }
       break;
     case REC_WRITE_TO_FLASH:
-      if (new_rec_state == REC_OFF || new_rec_state == REC_FILL_QUEUE) {
+      if ((new_rec_state == REC_OFF) || (new_rec_state == REC_FILL_QUEUE)) {
         rec_cmd = REC_CMD_WRITE_STOP;
-        osStatus_t ret = osMessageQueuePut(rec_cmd_queue, &rec_cmd, 0U, 10U);
+        osStatus_t ret = osMessageQueuePut(rec_cmd_queue, &rec_cmd, 0U, osWaitForever);
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/util/actions.c:244]: revert

File: boards/cats_rev1Pro/src/util/actions.c [hunk 4]
```diff
@@ -257,7 +259,7 @@ bool set_recorder_state(int16_t state) {
   global_recorder_status = new_rec_state;
 
   if (rec_cmd != REC_CMD_INVALID) {
-    osStatus_t ret = osMessageQueuePut(rec_cmd_queue, &rec_cmd, 0U, 10U);
+    osStatus_t ret = osMessageQueuePut(rec_cmd_queue, &rec_cmd, 0U, osWaitForever);
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/util/actions.c:262]: revert

**Review by jbinz — APPROVED** [review]

**Pull request merged**